system_tests/common_test.go

```diff
@@ -1733,7 +1733,11 @@ func recordBlock(t *testing.T, block uint64, builder *NodeBuilder) {
 			break
 		}
 	}
-	validationInputsWriter, err := inputs.NewWriter(inputs.WithSlug(t.Name()))
+	validationInputsWriter, err := inputs.NewWriter(
```

Comment:
I'm a little concerned that it is no longer even possible to include the test-specific slug or the blockId in the filename.

If you run two tests that record block inputs, the filenames will collide unless the test authors pass different baseDirectories (which previously they didn't have to worry about.)

Also, now, if a single test records two different block's validation inputs, they will be written to the same file, and the last one written will overwrite the previous ones. So, it won't be possible to consume separate files for the blocks any more.

I was really trying to make it as easy as possible for test authors to record blocks by handling collision avoidance in the inputs writer. But, now, to avoid the collisions every test author will have to make their baseDirectory unique for each block they record. I feel like that's a step in the wrong direction.  We could add flags (and again, I'd add them in this file rather than in the system_tests.go) to control whether or not a slug is used and whether or not the blockId is used in the filename. But, I think the default behavior should be to use both. Oh, and I'd also default to using the timestamp (because a user might run the same test in two different invocations and want the inputs from both.)

I understand that all this complexity in determining the filename makes the specific use case of wiring your CI test together a little more complicated. But, let's not remove the convenience provided for all the other use cases to make this one simpler to handle.

Reply:
I agree. My bad, my selfishness to make the CI wiring look simpler got the better of me :P
And thanks for this, I now understand the motivation for making inputswriter more customizable

Reply:
I've pushed the changes that makes this implementation backwards compatible